### PR TITLE
Fix line order bug

### PR DESCRIPTION
**Pull request by fosskers**
Previously I wasn't respecting the original Node order when construction `Line`s. This resulted in some exception throwing in OSMesa.

### CODE REVIEW

File: src/main/scala/vectorpipe/osm/Element.scala [hunk 1]
```diff
@@ -123,10 +124,11 @@ case class Member(
   ref: Long,
   role: String)
 
-case class ElementData(meta: ElementMeta, tagMap: Map[String, String])
+/** All Elements have common attributes, and may also have additional "tags". */
+@Lenses case class ElementData(meta: ElementMeta, tagMap: Map[String, String])
```
Comment (fosskers) [inline src/main/scala/vectorpipe/osm/Element.scala]: Auto-generated lenses are placed in the type's companion object.

File: src/main/scala/vectorpipe/osm/internal/ElementToFeature.scala [hunk 2]
```diff
@@ -283,7 +283,7 @@ private[vectorpipe] object ElementToFeature {
     case Vector()  => Vector.empty.pure[LineState]
     case Vector(h) => State.modify[List[OSMLine]](h :: _).map(_ => Vector.empty)
     case v => fuseOne(v) match {
-      case None => State.modify[List[OSMLine]](v.head :: _) >> fuseLines(v.tail)
+      case None => State.modify[List[OSMLine]](v.head :: _) *> fuseLines(v.tail)
```
Comment (fosskers) [inline src/main/scala/vectorpipe/osm/internal/ElementToFeature.scala]: Cats recently deprecated Monadic `>>` in favour of Applicative `*>`.

File: src/main/scala/vectorpipe/osm/internal/PlanetHistory.scala [hunk 3]
```diff
@@ -102,6 +93,26 @@ private[vectorpipe] object PlanetHistory {
...
+    f: Vector[Point] => G,
+    w: Way,
+    slices: List[(Instant, Map[Long, Node])]
+  ): List[Feature[G, ElementData]] = {
+    slices.foldLeft(Nil: List[Feature[G, ElementData]]) { case (acc, (i, ns)) =>
+      /* If a Node were deleted that the Way still expected, then no Line
+       * should be formed.
+       */
+      val points: Option[Vector[Point]] =
+        w.nodes.map(id => ns.get(id).map(n => Point(n.lon, n.lat))).sequence
```
Comment (fosskers) [inline src/main/scala/vectorpipe/osm/internal/PlanetHistory.scala]: This line in particular fixes the bug.

File: src/main/scala/vectorpipe/osm/internal/PlanetHistory.scala [hunk 4]
```diff
@@ -52,6 +53,9 @@ private[vectorpipe] object PlanetHistory {
       ps: List[OSMPolygon]
     ): (List[OSMLine], List[OSMPolygon]) = ws match {
       case Nil => (ls, ps)
+      /* OSM Extracts can produce degenerate Ways. These two checks guard against those. */
+      case w :: _ if w.nodes.length < 4 && w.isClosed => (ls, ps)
+      case w :: _ if w.nodes.length < 2 => (ls, ps)
```
Comment (fosskers) [inline src/main/scala/vectorpipe/osm/internal/PlanetHistory.scala]: Guard against degenerate Ways.

File: src/main/scala/vectorpipe/osm/Element.scala [hunk 5]
```diff
@@ -123,10 +124,11 @@ case class Member(
...
 
 /** All Element types have these attributes in common. */
-case class ElementMeta(
+@Lenses case class ElementMeta(
```
Comment (moradology) [inline src/main/scala/vectorpipe/osm/Element.scala]: I prefer this on a separate line, but I don't think this preference rises to the level of a standard

Comment (fosskers) [reply src/main/scala/vectorpipe/osm/Element.scala]: The Monocle author does it that way in his docs: https://julien-truffaut.github.io/Monocle/optics/lens.html

**Pull request merged**